src/selectors/entities/channel_categories.ts

```diff
@@ -91,36 +92,98 @@ export function makeFilterChannelsByType(): (state: GlobalState, channels: Chann
     );
 }
 
-export function makeFilterAutoclosedDMs(): (state: GlobalState, channels: Channel[], categoryType: string) => Channel[] {
+function getDefaultAutocloseCutoff() {
+    return Date.now() - 7 * 24 * 60 * 60 * 1000;
+}
+
+export function makeFilterAutoclosedDMs(getAutocloseCutoff = getDefaultAutocloseCutoff): (state: GlobalState, channels: Channel[], categoryType: string) => Channel[] {
     return createSelector(
         (state: GlobalState, channels: Channel[]) => channels,
         (state: GlobalState, channels: Channel[], categoryType: string) => categoryType,
-        (state: GlobalState) => state.entities.preferences.myPreferences,
+        getMyPreferences,
         shouldAutocloseDMs,
         getCurrentChannelId,
-        (channels, categoryType, myPreferences, autocloseDMs, currentChannelId) => {
+        (state: GlobalState) => state.entities.users.profiles,
+        getCurrentUserId,
+        getMyChannelMemberships,
+        getLastPostPerChannel,
+        (channels, categoryType, myPreferences, autocloseDMs, currentChannelId, profiles, currentUserId, myChannelMembers, lastPosts) => {
             if (categoryType !== CategoryTypes.DIRECT_MESSAGES) {
                 // Only autoclose DMs that haven't been assigned to a category
                 return channels;
             }
 
-            // TODO check if Preferences.CATEGORY_CHANNEL_APPROXIMATE_VIEW_TIME > now - 7 days to keep open
-            // TODO check if Preferences.CATEGORY_CHANNEL_OPEN_TIME > now - 7 days to keep open
-            // TODO check if teammate.delete_at > Preferences.CATEGORY_CHANNEL_OPEN_TIME to auto close if not current channel
-            // TODO check if lastPost.create_at > now - 7 days to keep open
-            // TODO check if channel.last_post_at > now - 7 days to keep open
-            // TODO don't autoclose favorited channel, except if teammate was deleted since last opened and not current channel
-            // TODO don't autoclose if autocloseDMs is false, except if teammate was deleted since last opened and not current channel
+            // Ideally, this would come from a selector, but that would cause the filter to recompute too often
+            const cutoff = getAutocloseCutoff();
```

Comment:
Nothing to act but This might be funny if user is looking at LHS and a DM just drops off because it exceeded.

Reply:
That's actually the current behaviour too. It just only happens if something else causes the selector to recalculate.